Improving the Design chapter

funkyfuture opened the pull request:
this should be merged into the main `branch` before the FAQ sections is added.

JKatzwinkel reviewed (approved):
Can't find any issues apart from not understanding what *Gestaltung* means in this context, and I also feel like there should maybe be a link to the design chapter in `index.rst` so that the reader proceeds to here first rather than being confronted with the quite heavy-weight web application serving the *getting started* guide, so I'm gonna push a small commit.

funkyfuture commented on docs/design.rst:
```diff
@@ -1,3 +1,5 @@
+.. _design:
```
i can't recall the proper role (maybe also `ref`), but you don't need to explicitly create an anchor inn order to reference documents or sections.

funkyfuture commented on docs/index.rst:
```diff
@@ -13,3 +13,6 @@
    genindex
    license
+
+For more in-depth reasoning about ``delb``'s philosophy, proceed to the
```
what about reordering the toc and making `design` the first item?

funkyfuture commented on docs/design.rst:
```diff
@@ -52,21 +52,23 @@ least these two share significant design aspects of Java APIs which is perceived
-they are formulated and structured for human consumers, and often printing
-devices.
+expectable. Text documents are different in these regards as natural languages
+and Gestaltung allow and lead to unprecedented manifestations for which an
```
see https://de.wikipedia.org/wiki/Gestaltung , afaik there's no english term for that. yes, it is Distinktionsjargon.

my point is that *layout* or *design* would be too narrow for the general contingency of *Textträgergestaltung*.

i think that's why Sachverhalte are often spelled out in english, but that is rather my ressentiment. anyway, that's an alternative.

funkyfuture commented:
lol, yeah sorry, i think there's even black for rst-snippets now.

that talk was shortly after the failed coup, wasn't it?

JKatzwinkel commented:
sry for the rant in cecc342, but I suffer severe OCD every time I encounter this stylistic choice, and hence crave the sympathy and affirmation of like-minded peers.

the talk was before the coup attempt, and it even says so on the background panel on stage, so I guess Henney and Stallman are off the hook on this one... I only found that video long after anyways, but it has held a special place in my heart since then because it tells me my opinions are good `:3`

JKatzwinkel commented:
should the design chapter have the caption *Design philosophy*?

funkyfuture commented on docs/design.rst:
```diff
@@ -97,14 +99,16 @@ concatenate broken words would need a function that removes the element objects
+   def remove_elements(
+       *elements: etree.ElementBase,
+       keep_children=False,
+       preserve_text=False,
+       preserve_tail=False
```
```suggestion
```

these do not apply with delb's data model.

funkyfuture commented on docs/design.rst:
```diff
@@ -97,14 +99,16 @@ concatenate broken words would need a function that removes the element objects
-                       preserve_text=False,
-                       preserve_tail=False) -> None:
+   def remove_elements(
+       *elements: etree.ElementBase,
```
```suggestion
       *nodes: delb.NodeBase,
```

JKatzwinkel replied:
isn't this the delbless example!?

funkyfuture replied:
no, no more at this point. it's about the same task with the same data type, but using a different data model.

funkyfuture replied:
oh no, this sparse-context view git me again. sorry, fixing it.

funkyfuture commented:
> should the design chapter have the caption _Design philosophy_?

i guess [so](https://www.youtube.com/watch?v=DlUyHogoyvE).

funkyfuture commented:
> the talk was before the coup attempt, and it even says so on the background panel on stage

nah, there's no "stay tuned, we're still planning…", but i'm impressed that a technical university bears an angel bee as mascot.

JKatzwinkel commented on docs/design.rst:
```diff
@@ -52,21 +52,23 @@ least these two share significant design aspects of Java APIs which is perceived
-they are formulated and structured for human consumers, and often printing
-devices.
+expectable. Text documents are different in these regards as natural languages
+and Gestaltung allow and lead to unprecedented manifestations for which an
```
I find this much easier to understand.

funkyfuture commented:
a'ight, i removed my mess, squashed all together and the only open question that remains is whether i mistook your proposal as teasing me over my jargon around the question of Gestaltung.
